kernel/src/actions/visitors.rs

```diff
@@ -1201,4 +1240,145 @@ mod tests {
             Some(1677811178585), // Retrieved ICT
         );
     }
+
+    // Empty mock GetData for testing error cases where we need wrong number of getters
+    struct MockGetter;
+    impl<'a> GetData<'a> for MockGetter {}
+
+    #[test]
+    fn test_add_visitor_wrong_getter_count() {
+        let json_strings: StringArray = vec![add_action()].into();
+        let batch = parse_json_batch(json_strings);
+
+        // Verify normal operation with correct number of getters
+        let mut visitor = AddVisitor::default();
+        visitor.visit_rows_of(batch.as_ref()).unwrap();
+        assert_eq!(visitor.adds.len(), 1);
+
+        // Test error case: call visit_add with only 5 getters instead of required 15
+        let mock = MockGetter;
+        let wrong_getters: Vec<&dyn GetData<'_>> = vec![&mock, &mock, &mock, &mock, &mock];
+
+        let result = AddVisitor::visit_add(0, "test.parquet".to_string(), &wrong_getters);
+
+        assert!(result.is_err());
+        let err = result.unwrap_err();
+        assert!(
+            matches!(
+                &err,
+                VisitorError::InvalidGetterCount {
+                    action_type,
+                    expected,
+                    actual,
+                } if action_type == "Add" && *expected == 15 && *actual == 5
+            ),
+            "Expected InvalidGetterCount with Add/15/5, got {:?}",
```

Comment:
Nit: the slashes Add/15/5 seem like strange short hand can these be spelled out?